### ISSUE TITLE
Issues with some query builder methods on relation

### ISSUE THREAD

**Issue by canvural** [nunomaduro/larastan#411]
- Larastan Version: master
- `--level` used: 7

### Description

After #410 we have some problems with some of the query builder methods called on relations.

For example `increment` ;

```php
$accountTransaction->account()->increment('balance', 10);
```

gives the following error
```
Method Illuminate\Database\Eloquent\Model::increment() invoked with 2 parameters, 0 required.
```

Or any of the retrieval methods, like `first`

```php
return $this->hasOne(BookingEvent::class)->oldest()->orderBy('id')->first();
```

gives the following error

```
Method App\Models\xxx::xxx() should return App\Models\xxx|null but returns Illuminate\Database\Eloquent\Relations\HasOne.
```

I think this can be fixed by using a similar logic used in [here](https://github.com/nunomaduro/larastan/blob/master/src/Methods/ModelForwardsCallsExtension.php#L79-L91). Maybe we can extract this piece of code to another helper, like `BuilderHelper` But in order to make this change, we have to know which model would be returned. I will work on this sometime soon. Opened the issue to just track it.

**Comment by BertvanHoekelen** [comment]
Let me know if I can be of any help! I can definitely start working on some more relation fixes once I have access to the related model.

**Comment by canvural** [comment]
Fixed in #419 and #423
